src/Microsoft.OData.Edm/EdmModelVisitor.cs

```diff
@@ -427,21 +455,45 @@ protected virtual void ProcessElement(IEdmElement element)
             this.VisitAnnotations(this.Model.DirectValueAnnotations(element));
         }
 
+        protected virtual Task ProcessElementAsync(IEdmElement element)
+        {
+            // TODO: DirectValueAnnotationsInMainSchema (not including those in referenced schemas)
+            this.VisitAnnotations(this.Model.DirectValueAnnotations(element));
+
+            return Task.FromResult(0);
+        }
+
         protected virtual void ProcessNamedElement(IEdmNamedElement element)
         {
             this.ProcessElement(element);
         }
 
+        protected virtual async Task ProcessNamedElementAsync(IEdmNamedElement element)
+        {
+            await this.ProcessElementAsync(element);
+        }
+
         protected virtual void ProcessSchemaElement(IEdmSchemaElement element)
         {
             this.ProcessVocabularyAnnotatable(element);
             this.ProcessNamedElement(element);
         }
 
+        protected virtual async Task ProcessSchemaElementAsync(IEdmSchemaElement element)
+        {
+            await this.ProcessVocabularyAnnotatableAsync(element);
+            await this.ProcessNamedElementAsync(element);
+        }
+
         protected virtual void ProcessVocabularyAnnotatable(IEdmVocabularyAnnotatable annotatable)
         {
         }
 
+        protected virtual Task ProcessVocabularyAnnotatableAsync(IEdmVocabularyAnnotatable annotatable)
+        {
+            return Task.FromResult(0);

```

Comment:
Why does this return `Task.FromResult(0)` and not `Task.CompletedTask` given that the declared return type is not int?